pkg/pool-stable-phantom/test/StablePhantomPool.test.ts

```diff
@@ -968,237 +971,130 @@ describe('StablePhantomPool', () => {
-          // In StablePools, BPT and underlying tokens are almost equivalent. This means that the token fee amount is a
-          // good estimate of the equivalent BPT fee amount.
+          sharedBeforeEach('update cache', async () => {
+            await pool.updateProtocolSwapFeePercentageCache();
+            inRecoveryMode = await pool.inRecoveryMode();
+          });
 
-          if (kind == AmountKind.WITHOUT_FEE) {
-            amount = amount.mul(fp(1)).div(fp(1).sub(swapFeePercentage));
+          enum AmountKind {
+            WITH_FEE,
+            WITHOUT_FEE,
           }
 
-          const fee = amount.mul(swapFeePercentage).div(fp(1));
-          const protocolFee = fee.mul(protocolFeePercentage).div(fp(1));
-          return protocolFee;
-        }
-
-        context('on swaps given in', () => {
-          it('tracks fees when swapping tokens', async () => {
-            const previousDueFee = await pool.getDueProtocolFeeBptAmount();
+          function getExpectedProtocolFee(amount: BigNumber, kind: AmountKind, recoveryMode: boolean): BigNumber {
+            // In StablePools, BPT and underlying tokens are almost equivalent. This means that the token fee amount is a
+            // good estimate of the equivalent BPT fee amount.
 
-            const tokenIn = tokens.first;
-            const tokenOut = tokens.second;
-            const { receipt } = await pool.swapGivenIn({ in: tokenIn, out: tokenOut, amount, from: lp, recipient });
+            if (recoveryMode) {
+              return bn(0);
+            }
 
-            const currentDueFee = await pool.getDueProtocolFeeBptAmount();
-            const aproxFee = getAproxDueFee(amount, AmountKind.WITH_FEE);
+            if (kind == AmountKind.WITHOUT_FEE) {
+              amount = amount.mul(fp(1)).div(fp(1).sub(swapFeePercentage));
+            }
 
-            expect(currentDueFee).to.be.equalWithError(aproxFee, 0.01);
+            const fee = amount.mul(swapFeePercentage).div(fp(1));
+            const protocolFee = fee.mul(protocolFeePercentage).div(fp(1));
+            return protocolFee;
+          }
 
-            expectEvent.inIndirectReceipt(receipt, pool.instance.interface, 'DueProtocolFeeIncreased', {
-              bptAmount: currentDueFee.sub(previousDueFee),
-            });
-          });
+          context('on swaps given in', () => {
+            it('pays any protocol fees due when swapping tokens', async () => {
+              const tokenIn = tokens.first;
+              const tokenOut = tokens.second;
+              await pool.swapGivenIn({ in: tokenIn, out: tokenOut, amount, from: lp, recipient });
 
-          it('tracks fees when swapping for BPT (join)', async () => {
-            const previousDueFee = await pool.getDueProtocolFeeBptAmount();
+              const currentFee = await pool.balanceOf(protocolFeesCollector.address);
+              const expectedFee = getExpectedProtocolFee(amount, AmountKind.WITH_FEE, inRecoveryMode);
```

Comment:
This is a bit fragile as it relies on the initial fee collector balance to be zero. Can we change these so that they operate on deltas instead?

Reply:
Will do